### DISCUSSION TITLE
让组件、插件的开发脱离该仓库

### DISCUSSION THREAD

**Discussion by FoundTheWOUT** [the1812/Bilibili-Evolved#3139] [综合讨论 / General]
### 想法
最近在考虑能不能让组件和插件脱离开本仓库进行开发（并做到开发友好），先把想法以及一些进程分享出来，看下如何，以及有没有研究下去的意义。~~免得浪费时间~~
#### 想了下大概有这几件事要做
1. 生成 dts 文件
    opts: 将 dts 文件打包成单个
2. 发布类型文件
3. 可能的话做一个 template，或者一个 cli
### 一些进程
关于第一点做了下研究以及尝试。
设置 `package.json` 里的配置
```
  "main": "dist/src/core/core-apis.js",
  "types": "dist/src/core/core-apis.d.ts",
```
给 vscode 提供类型支持。
![image](https://user-images.githubusercontent.com/47922588/158964638-1f30b4be-d0fa-4bbe-a6f7-4f39c0f2d33c.png)

也可以利用[API Extractor](https://api-extractor.com/)把dts打包成一个文件， types 指向打包好的文件。但现阶段该工具不能完美的用这个项目里用起来。一些语法并未支持，比如 `typeof import('/path')` 。

### 其它
公共UI问题等等

**Comment by the1812** [comment]
有必要分离吗, 看起来要做的事情只是在还原分离前的功能.
组件、插件的开发我认为基于分支就可以了, 合在一个仓库里就是为了能够无缝同步各种依赖的数据和类型

**Reply by FoundTheWOUT** [reply]
这也不无道理
